webapp/tests/forms/flows/test_lotse_flow.py

```diff
@@ -1037,6 +1038,19 @@ def test_if_familienstand_step_then_delete_gem_haushalt_correctly(self):
             self.assertNotIn('stmind_gem_haushalt_entries', returned_data)
             self.assertNotIn('stmind_gem_haushalt_count', returned_data)
 
+        # separated
+        with self.app.test_request_context(method='POST',
+                                           data={'familienstand': 'married',
+                                                 'familienstand_date': ['1', '1', '1985'],
+                                                 'familienstand_married_lived_separated': 'yes',
+                                                 'familienstand_married_lived_separated_since': ['1', '1', '1990']}):
+            _, returned_data = self.flow._handle_specifics_for_step(
+                self.familienstand_step, self.render_info_familienstand_step,
+                {'stmind_gem_haushalt_entries': ['Helene Fischer'], 'stmind_gem_haushalt_count': 1})
+            self.assertIn('stmind_gem_haushalt_entries', returned_data)
+            self.assertIn('stmind_gem_haushalt_count', returned_data)
```

Comment:
Imo we do not need this and instead could rename the test or we have two tests that both only test one thing (delete if Zusammenveranlagung and delete not if not Zusammenveranlagung. We could also think about patching show_person_b